opengrok-tools/src/main/python/opengrok_tools/utils/restful.py

```diff
@@ -38,14 +38,24 @@ def do_api_call(verb, uri, params=None, headers=None, data=None):
     if handler is None or not callable(handler):
         raise Exception('Unknown HTTP verb: {}'.format(verb))
 
-    return handler(
+    logger = logging.getLogger(__name__)
+
+    logger.debug("{} API call: {} with data '{}' and headers: {}".
+                 format(verb, uri, data, headers))
+    r = handler(
         uri,
         data=data,
         params=params,
         headers=headers,
         proxies=get_proxies(uri)
     )
 
+    if r is not None:
+        logger.error("API call failed: {}".format(r))
+        r.raise_for_status()
```

Comment:
I'd interpret this behavior to the doc string (exception on failed requests)

Reply:
added a doc string